test/t8_schemes/t8_gtest_element_ref_coords.cxx

```diff
@@ -0,0 +1,189 @@
+  double centroid_by_element_ref_coords
+    [3]; /** centroid computed via \ref t8_forest_element_centroid (uses \ref t8_element_reference_coords) -> \ref t8_geometry_evaluate */
+  double batch_coords
+    [T8_ECLASS_MAX_CORNERS
+     * T8_ECLASS_MAX_DIM]; /** reference coordinates of the element vertices computed by \ref t8_get_batch_coords_for_element_type */
+
+  /* compare results of the two different way to obtain tree ref coords */
+  const t8_element_shape_t shape = ts->t8_element_shape (element);
+  const int num_vertices = t8_eclass_num_vertices[shape];
+  const int elem_dim = t8_eclass_to_dimension[shape];
+  t8_get_batch_coords_for_element_type (shape, batch_coords);
+  t8_debugf ("Testing for shape %s\n", t8_eclass_to_string[shape]);
+  t8_debugf ("with num_vertices %i\n", num_vertices);
+  t8_debugf ("and elem_dim %i\n", elem_dim);
+
+  ts->t8_element_reference_coords (element, batch_coords, num_vertices, tree_ref_coords_by_element_ref_coords);
+  for (int i_vertex = 0; i_vertex < num_vertices; ++i_vertex) {
+    ts->t8_element_vertex_reference_coords (element, i_vertex, tree_ref_coords_by_vertex);
+    t8_debugf ("Checking vertex %i\n", i_vertex);
+    t8_write_message_by_dim ("with the batch coords:", batch_coords, elem_dim);
+    t8_write_message_by_dim ("tree_ref_coords_by_vertex:", tree_ref_coords_by_vertex, elem_dim);
+    t8_write_message_by_dim (
+      "tree_ref_coords_by_element_ref_coords:", tree_ref_coords_by_element_ref_coords + i_vertex * elem_dim, elem_dim);
+    for (int dim = 0; dim < elem_dim; ++dim) {
+      EXPECT_NEAR (tree_ref_coords_by_vertex[dim], tree_ref_coords_by_element_ref_coords[i_vertex * elem_dim + dim],
+                   2 * T8_PRECISION_EPS);
+    }
+  }
+  /* compare results of the two different ways to compute an elements centroid */
+  t8_forest_element_centroid (forest, ltree_id, element, centroid_by_element_ref_coords);
+  t8_element_centroid_by_vertex_coords (forest, ts, ltree_id, element, centroid_by_vertices);
+  t8_write_message_by_dim ("centroid_by_vertices:", centroid_by_vertices, 3);
+  t8_write_message_by_dim ("centroid_by_element_ref_coords:", centroid_by_element_ref_coords, 3);
+  for (int dim = 0; dim < T8_ECLASS_MAX_DIM; ++dim) {
+    EXPECT_NEAR (centroid_by_vertices[dim], centroid_by_element_ref_coords[dim], 2 * T8_PRECISION_EPS);
+  }
+
+  return 0;
+}
+
+class class_ref_coords: public testing::TestWithParam<t8_eclass> {
+ protected:
+  void
+  SetUp () override
+  {
+    eclass = GetParam ();
+    cmesh = t8_cmesh_new_from_class (eclass, sc_MPI_COMM_WORLD);
+  }
+  void
+  TearDown () override
+  {
+    t8_cmesh_unref (&cmesh);
+  }
+  t8_eclass_t eclass;
+  t8_cmesh_t cmesh;
+  t8_forest_t forest, forest_partition;
+};
+
+TEST_P (class_ref_coords, t8_check_elem_ref_coords)
+{
+  t8_locidx_t itree, ielement;
+  /* Generate a uniform forest for each level up to MAXLEVEL and partition it */
+  for (int level = 0; level <= MAXLEVEL; ++level) {
```

Comment:
From 0 to MaxLevel is actually a big range. How much time does the test need?

Can you add level as a parameter to the testsuite? Then you can move the construction of the forest also into the constructor of the class.

Reply:
Ah, you can ignore my comment. I thaught this was the maximal allowed level for that element. Can you rename it to something like `MAX_LEVEL_REF_COORD_TEST`

Reply:
Done

Reply:
That was ambiguous. I still think making the level a parameter of the testsuite is a good idea ;)

Sorry for not being clear about it.

Reply:
Done